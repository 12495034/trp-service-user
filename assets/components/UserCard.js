import React from 'react'
import { StyleSheet, View, Image } from 'react-native';
import { Text } from 'react-native-paper';
import { Linking } from 'react-native';

/**
 * User card component renders personal details entered by the user at signup
 */
export default function UserCard(props) {
    return (
        <View style={UserCardStyles.body}>
            <View style={UserCardStyles.title}>
                <Text style={UserCardStyles.titleText}>{props.email}</Text>
            </View>
            <View style={UserCardStyles.userInfo}>
                <View>
                    <Text style={UserCardStyles.detailText}>Pronouns:</Text>
                    <Text style={UserCardStyles.detailText}>First Name:</Text>
                    <Text style={UserCardStyles.detailText}>Middle Name:</Text>
                    <Text style={UserCardStyles.detailText}>Last Name:</Text>
                    <Text style={UserCardStyles.detailText}>Phone Number:</Text>
                    <Text style={UserCardStyles.detailText}>Date of Birth:</Text>
                    <Text style={UserCardStyles.detailText}>Email Verified:</Text>
                    <Text style={UserCardStyles.detailText}>Created On:</Text>
                    <Text style={UserCardStyles.detailText}>Notifications:</Text>
                    <Text style={UserCardStyles.detailText}>Status:</Text>
                    <Text></Text>
                    <Text
                        style={{ color: 'blue', textDecorationLine: 'underline', }}
                        onPress={() => Linking.openURL('https://trp-developement.web.app/privacy-policy-mobile')}>
                        View Privacy Policy
                    </Text>

                </View>
                <View>
                    <Text style={UserCardStyles.detailText}>{props.proNouns}</Text>
                    <Text style={UserCardStyles.detailText}>{props.firstName}</Text>
                    <Text style={UserCardStyles.detailText}>{props.middleName}</Text>
                    <Text style={UserCardStyles.detailText}>{props.lastName}</Text>
                    <Text style={UserCardStyles.detailText}>{props.phoneNumber}</Text>
                    <Text style={UserCardStyles.detailText}>{props.dob}</Text>
                    <Text style={UserCardStyles.detailText}>{props.emailVerified ? "Yes" : "No"}</Text>
                    <Text style={UserCardStyles.detailText}>{props.creationDate}</Text>
                    <Text style={UserCardStyles.detailText}>{props.emailOptIn ? "Yes" : "No"}</Text>
                    <Text style={UserCardStyles.detailText}>{props.status}</Text>
                </View>
            </View>
        </View>
    )
}

const UserCardStyles = StyleSheet.create({
    body: {
        flex: 1,
        width: '100%',
        borderWidth: 1,
        borderStyle: 'solid',
        borderRadius: 10,
        marginBottom: 5,
    },
    image: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        padding: 10,
    },
    userInfo: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'space-between',
        padding: 20,
    },
    title: {
        padding: 5,
    },
    titleText: {
        textAlign: 'center',
        fontSize: 18,
        color: 'blue',
        backgroundColor: '#F2D6F3',
        padding: 10,
        borderRadius: 5,
        elevation: 5,
    },
    detailText: {
        fontSize: 15,
        paddingBottom: 2,
        color: 'black',
    }
})
