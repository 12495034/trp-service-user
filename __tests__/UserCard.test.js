import React from 'react';
import renderer from 'react-test-renderer';
import UserCard from '../assets/components/UserCard';

describe('User Card Tests', () => {
    test('renders userCard component correctly wuth true values', () => {
        const tree = renderer.create(<UserCard
            proNouns={'He/Him'}
            firstName={'Gavin'}
            middleName={'James'}
            lastName={'Davis'}
            dob={'1983-01-01'}
            email={'testEmail@Email.com'}
            emailVerified={true}
            phoneNumber={123456}
            isAgreedTC={true}
        />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('renders userCard component correctly with false values', () => {
        const tree = renderer.create(<UserCard
            proNouns={'He/Him'}
            firstName={'Gavin'}
            middleName={'James'}
            lastName={'Davis'}
            dob={'1983-01-01'}
            email={'testEmail@Email.com'}
            emailVerified={false}
            phoneNumber={123456}
            isAgreedTC={false}
        />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});