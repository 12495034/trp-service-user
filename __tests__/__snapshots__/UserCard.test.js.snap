// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Card Tests renders userCard component correctly with false values 1`] = `
<View
  style={
    {
      "borderRadius": 10,
      "borderStyle": "solid",
      "borderWidth": 1,
      "flex": 1,
      "marginBottom": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "padding": 5,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          {
            "textAlign": "left",
          },
          {
            "backgroundColor": "#F2D6F3",
            "borderRadius": 5,
            "color": "blue",
            "elevation": 5,
            "fontSize": 18,
            "padding": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      testEmail@Email.com
    </Text>
  </View>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 20,
      }
    }
  >
    <View>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Pronouns:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        First Name:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Middle Name:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Last Name:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Phone Number:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Date of Birth:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Email Verified:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Created On:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Notifications:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Status:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            undefined,
          ]
        }
      />
      <Text
        onPress={[Function]}
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "blue",
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        View Privacy Policy
      </Text>
    </View>
    <View>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        He/Him
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Gavin
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        James
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Davis
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        123456
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        1983-01-01
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        No
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      />
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        No
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`User Card Tests renders userCard component correctly wuth true values 1`] = `
<View
  style={
    {
      "borderRadius": 10,
      "borderStyle": "solid",
      "borderWidth": 1,
      "flex": 1,
      "marginBottom": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "padding": 5,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          {
            "textAlign": "left",
          },
          {
            "backgroundColor": "#F2D6F3",
            "borderRadius": 5,
            "color": "blue",
            "elevation": 5,
            "fontSize": 18,
            "padding": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      testEmail@Email.com
    </Text>
  </View>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 20,
      }
    }
  >
    <View>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Pronouns:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        First Name:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Middle Name:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Last Name:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Phone Number:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Date of Birth:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Email Verified:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Created On:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Notifications:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Status:
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            undefined,
          ]
        }
      />
      <Text
        onPress={[Function]}
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "blue",
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        View Privacy Policy
      </Text>
    </View>
    <View>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        He/Him
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Gavin
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        James
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Davis
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        123456
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        1983-01-01
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        Yes
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      />
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      >
        No
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "color": "black",
              "fontSize": 15,
              "paddingBottom": 2,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
